add "Lint your tag files" rule

jbmoelker opened the pull request:
## Summary

Proposed changes:
- add "Lint your tag files" rule (incl. summary, why, how)
- instructions for configuring and running ESLint & JSHint

Resolves #32.
## Checklist
- [X] The changes only affect or contain one style guide rule or section.
- [X] The changes are in [style guide format](/CONTRIBUTING.md#format).
- [X] The new style guide section has an entry in the [Table of Contents](/README.md#table-of-contents) (only if changes introduce new section).
- [X] The changes can be merged into the target branch without conflicts.

jbmoelker committed:
add "Lint your tag files" rule

incl. summary, why, how and instructions for ESLint & JSHint

closes #32

jbmoelker commented:
@phortuin I know you were against this in #32. I'm hoping you'll reconsider. I do think it benefits teams. The AngularJS Style Guide also has a [JS Hint section](https://github.com/johnpapa/angular-styleguide/tree/master/a1#js-hint) (and JSCS, no ESLint). 

I've also tested both ESLint and JSHint in a Riot project: [ESLint setup](https://github.com/jbmoelker/demo-viewer/commit/b864517144ea9b90ecb249b0d194589dee4d0b7f), [JSHint setup](https://github.com/jbmoelker/demo-viewer/commit/8c93ca3c46fe480701d44a91d5c46c973faae359). 

Maybe our pro linters @jpsc and @petergoes can also review :).

phortuin commented on README.md:
```diff
@@ -718,3 +719,61 @@ Example demo file in `city-tag` module:
+
+## How?
+
+To allow linters to extract the scripts from your `*.tag.html` files, [put script inside a `<script>` tag](#use-script-inside-tag) and [keep tag expressions simple](#keep-tag-expressions-simple) (as linters don't understand those). Then configure your linter to allow `riot` and tag `opts`.
```
Wording: "Configure your linter to allow global variables `riot` and `opts`."

jbmoelker committed:
rephrase to "configure to allow global variables"

resolves comment https://github.com/voorhoede/riotjs-style-guide/pull/39#discussion_r57699402

jbmoelker commented:
@phortuin I agree on https://github.com/voorhoede/riotjs-style-guide/pull/39#discussion-diff-57699402. Updated in https://github.com/voorhoede/riotjs-style-guide/pull/39/commits/03230874e785ecc5d907c4c0949645de6e2c4e1e.
